### DISCUSSION TITLE
When the page changes, express doesn't work, getServerSideProps works.

### DISCUSSION THREAD

**Discussion by phibersoft** [vercel/next.js#15074] [Help]
Hi first,

I will explain my problem through an example.

I have a three components under the /pages.
/a and /b and /index
Example a : 
 `class A extends React.Component(...render...) 
export async function getServerSideProps(ctx){ console.log("ServerSideProps running") }`

Im using nextjs's Custom Server. (More information : https://nextjs.org/docs/advanced-features/custom-server)

example my server : 
 `
 server.get("/a", async(req,res) => {
  console.log("Server side express running");
app.render(req,res,"/a",req.query);
  })
`

OUTPUT : 
Server side express running,
ServerSideProps running

--------
So far, no problem. But if I come back to page "b" after rendering a, then back to "a" page, the output looks like this:

OUTPUT:
Server side express running.
ServerSideProps running
// page changed /b and redirect again /a
ServerSideProps running

Is it a bug or my fault?

**Comment by surferwat** [comment]
@phibersoft What are you expecting the output to look like?

**Reply by phibersoft** [reply]
### Expected Output
Server side express running // path a
ServerSide props running // path a
// Page changed with next/link (/a to /b)
Server side express running // path b
ServerSide props running // path b
// Page changed with next/link (/b to /a)
Server side express running // path a
ServerSide props running // path a

### Received Output
Server side express running // path a
ServerSide props running // path a 
// Page changed with next/link (/a to /b)
ServerSide props running // path b
// Page changed with next/link (/b to /a)
ServerSide props running // path a
